File: server/packets/client/PacketClientWalk.cs
using System.Text.Json.Serialization;

public class PacketClientWalk : Packet {

    [JsonPropertyName("speed")]
    public double Speed { get; set; }

    [JsonPropertyName("direction")]
    public byte Direction { get; set; }

    [JsonPropertyName("x")]
    public int X { get; set; }

    [JsonPropertyName("y")]
    public int Y { get; set; }
}

File: src/network/packets/client/PacketClientWalk.ts
import Direction from "../../../render/Direction.ts";
import Packet from "../Packet.ts";

export default class PacketClientWalk implements Packet {

    packetType: string;
    direction: Direction;
    speed: number;
    x: number;
    y: number;

    constructor(direction: Direction, speed: number, x: number, y: number) {
        this.packetType = "PacketClientWalk";
        this.direction = direction;
        this.speed = speed;
        this.x = x;
        this.y = y;
    }

}